From 7f457bc1b4f20d66644ecb9c8fda39cfc6a1b86d Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 3 Feb 2021 00:10:14 +0000
Subject: [PATCH 4/5] gnu: Add printv

* gnu/packages/lisp-xyz.scm
  (sbcl-printv, cl-printv, ecl-printv): New variables
---
 gnu/packages/lisp-xyz.scm | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index e5f230214e..02fb92b9a3 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -4731,6 +4731,36 @@ various levels and mix text with expressions.")
 (define-public ecl-log4cl
   (sbcl-package->ecl-package sbcl-log4cl))
 
+(define-public sbcl-printv
+  (let ((commit "646d31978dbbb460fffb160fd65bb2be9a5a434e")
+        (revision "1"))
+    (package
+      (name "sbcl-printv")
+      (version (git-version "0.1.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/danlentz/printv")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "08jvy82abm7qi3wrxh6gvmwg9gy0zzhg4cfqajdwrggbah8mj5a6"))))
+      (build-system asdf-build-system/sbcl)
+      (home-page "https://github.com/danlentz/printv")
+      (synopsis "Common Lisp tracing and debug-logging macro")
+      (description
+       "@code{PRINTV} is a \"batteries-included\" tracing and debug-logging macro.
+
+This package provides 1 system: @code{PRINTV}")
+      (license license:asl2.0))))
+
+(define-public ecl-printv
+  (sbcl-package->ecl-package sbcl-printv))
+
+(define-public cl-printv
+  (sbcl-package->cl-source-package sbcl-printv))
+
 (define-public sbcl-verbose
   (let ((commit "c5b7ecd465be61b35af17ef57564697b88397174")
         (revision "1"))
-- 
2.30.0

